### PR TITLE
Fix groupby.cumsum with series grouped by index

### LINKED ISSUE #8578: groupBy on index of Series fails
**Minimal Complete Verifiable Example**:

```python
import numpy as np
import pandas as pd
import dask.dataframe as dd

df = pd.DataFrame(np.asarray([[0,3,2],[0,2,8],[1,3,8]]), columns=["A","B","C"]).set_index("A")

# expected
print(df["B"].groupby("A").cumsum())

# In dask
ddf = dd.from_pandas(df, npartitions=1)
print(ddf["B"].groupby("A").cumsum().compute())
```

Instead of returning the same result as plain pandas, the above snippet crashes with the following traceback:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_4276/396749366.py in <module>
      1 # real
      2 ddf = dd.from_pandas(df, npartitions=1)
----> 3 ddf["B"].groupby("A").cumsum().compute()

.../python3.9/site-packages/dask/dataframe/groupby.py in cumsum(self, axis)
   1339             return self.obj.cumsum(axis=axis)
   1340         else:
-> 1341             return self._cum_agg("cumsum", chunk=M.cumsum, aggregate=M.add, initial=0)
   1342 
   1343     @derived_from(pd.core.groupby.GroupBy)

.../python3.9/site-packages/dask/dataframe/groupby.py in _cum_agg(self, token, chunk, aggregate, initial)
   1230 
   1231         cumpart_ext = cumpart_raw_frame.assign(
-> 1232             **{
   1233                 i: self.obj[i]
   1234                 if np.isscalar(i) and i in self.obj.columns

.../python3.9/site-packages/dask/dataframe/groupby.py in <dictcomp>(.0)
   1232             **{
   1233                 i: self.obj[i]
-> 1234                 if np.isscalar(i) and i in self.obj.columns
   1235                 else self.obj.index
   1236                 for i in by

AttributeError: 'Series' object has no attribute 'columns'

```

The above computation works fine when grouping the full dataframe, for eg `ddf.groupby("A").cumsum().compute()`
**Environment**:

- Dask version: '2022.01.0'
- Python version: 3.9
- Operating System: linux
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by jsignell**
- [x] Closes #8578
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

I need some help figuring out what is making the test fail - maybe @ian-r-rose can provide a second set of eyes. 

Here's an example of the failure:

```python-traceback
_____________________ test_series_groupby_with_named_index[disk] ______________________

    def test_series_groupby_with_named_index():
        df = pd.DataFrame({"x": [1, 2, 3], "y": [4, 5, 6]}).set_index("x")
        ddf = dd.from_pandas(df, 2)
    
        expected = df["y"].groupby("x").cumsum()
        result = ddf["y"].groupby("x").cumsum()
        breakpoint()
    
>       assert_eq(result, expected)

dask/dataframe/tests/test_groupby.py:441: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dask/dataframe/utils.py:543: in assert_eq
    a = _check_dask(
dask/dataframe/utils.py:453: in _check_dask
    graph.validate()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = HighLevelGraph with 8 layers.
<dask.highlevelgraph.HighLevelGraph object at 0x7fb7a0f66790>
 0. from_pandas-67819a5ba8...s-groupby-cumsum-take-last-5afb376059afe4fed9ae470398b312d5
 7. series-groupby-cumsum-c2a355e10b84617b49c2c62c6d084b7a


    def validate(self):
        # Check dependencies
        for layer_name, deps in self.dependencies.items():
            if layer_name not in self.layers:
                raise ValueError(
                    f"dependencies[{repr(layer_name)}] not found in layers"
                )
            for dep in deps:
                if dep not in self.dependencies:
                    raise ValueError(f"{repr(dep)} not found in dependencies")
    
        for layer in self.layers.values():
            assert hasattr(layer, "annotations")
    
        # Re-calculate all layer dependencies
        dependencies = compute_layer_dependencies(self.layers)
    
        # Check keys
        dep_key1 = self.dependencies.keys()
        dep_key2 = dependencies.keys()
        if dep_key1 != dep_key2:
            raise ValueError(
                f"incorrect dependencies keys {set(dep_key1)!r} "
                f"expected {set(dep_key2)!r}"
            )
    
        # Check values
        for k in dep_key1:
            if self.dependencies[k] != dependencies[k]:
>               raise ValueError(
                    f"incorrect dependencies[{repr(k)}]: {repr(self.dependencies[k])} "
                    f"expected {repr(dependencies[k])}"
                )
E               ValueError: incorrect dependencies['series-groupby-cumsum-c2a355e10b84617b49c2c62c6d084b7a']: {'assign-e925ccbf6d5a86174f1180eea0696a25', 'series-groupby-cumsum-take-last-5afb376059afe4fed9ae470398b312d5', 'series-groupby-cumsum-map-797524ad685aa1db670af58d96e53cc2'} expected {'series-groupby-cumsum-map-797524ad685aa1db670af58d96e53cc2'}

dask/highlevelgraph.py:1035: ValueError

```

### CODE REVIEW

**Comment by ian-r-rose** [comment]
Sorry for the delay @jsignell, I took a look at this this afternoon. There are a couple of things going on here, but at heart: the graph validation step is complaining that the HighLevelGraph layer dependencies aren't what we say they are. In particular, the issue lies with this block: 
https://github.com/dask/dask/blob/4228dc7439f4ecd05708641bbe25adbf4d658f92/dask/dataframe/groupby.py#L1263-L1284

If the dataframe is a single partition, then the `dsk` never gets any dependencies on `cumpart_ext` or `cumlast`. But in `HighLevelGraph.from_collections` we are saying that it does have those dependencies. I think a relatively straightforward fix could be to do something like:
```python
dependencies = [cumpart_raw]
if self.obj.npartitions > 1:
    dependencies += [cumpart_ext, cumlast]
```

Now, I see that the test explicitly asks for two partitions, so I'm not sure why `from_pandas` is giving you a single partition dataframe. But hopefully the above should unblock this PR?

File: dask/dataframe/tests/test_groupby.py [hunk 1]
```diff
@@ -430,6 +430,14 @@ def test_series_groupby_propagates_names():
     assert_eq(result, expected)
 
 
+def test_series_groupby_with_named_index():
```
Comment (ian-r-rose) [inline dask/dataframe/tests/test_groupby.py:433]: Given that there are different code paths for single-partition and multi-partition dataframes, perhaps we should parameterize over `npartitions`? Could also parameterize over `cumprod` and `cumcount` while we are at it

**Comment by ian-r-rose** [comment]
N.b., I was able to get the test passing by extending the length of your test dataframe, which was enough to convince `from_pandas` to give us a two-partition ddf.

**Comment by jsignell** [comment]
Thank you so much for the review Ian! This all makes a lot of sense.

**Review by ian-r-rose — APPROVED** [review]
Thanks @jsignell!

**Pull request merged**